code/design_pattern/strategy_pattern/strategy_pattern.cpp

```diff
@@ -0,0 +1,141 @@
+#include <iostream>
+using namespace std;
+
+class IFlyBehavior
+{
+	public :
+	virtual ~IFlyBehavior(){}
+	virtual void fly() = 0;
+};
+
+class FlyWithWings : public IFlyBehavior
+{
+	public :
+	void fly()
+	{
+		cout << "I fly with wings!" << endl;
+	}	
+};
+
+class FlyNoWay : public IFlyBehavior
+{
+	public :
+	void fly() 
+	{
+		cout << "I don't fly, buddy." << endl;
+	}
+};
+
+class IQuackBehavior 
+{
+	public :
+	virtual ~IQuackBehavior(){}
+	virtual void quack() = 0;
+};
+
+class Quack : public IQuackBehavior
+{
+	public :
+	void quack()
+	{
+		cout << "Quack!" << endl;
+	}
+};
+
+class Squeak : public IQuackBehavior
+{
+	public :
+	void quack() 
+	{
+		cout << "Squeak!" << endl;
+	}
+};
+
+class MuteQuack : public IQuackBehavior
+{
+	public :
+	void quack()
+	{
+		cout << "I'm a mute duck." << endl;
+	}
+};
+
+class Duck 
+{
+	public :
+	IFlyBehavior* flyBehavior;
+	IQuackBehavior* quackBehavior;
```

Comment:
Also, since these are dynamic pointers, either add a destructor to delete these objects, or turn them into smart pointers. We don't want memory leaks in this code!